File: src/Java/testapp/src/main/java/com/myservicebus/testapp/dashboard/DashboardApi.java
package com.myservicebus.testapp.dashboard;

import com.myservicebus.inspection.BusInspectionProvider;
import io.javalin.Javalin;

public final class DashboardApi {
    private DashboardApi() {
    }

    public static void register(Javalin app, BusInspectionProvider inspectionProvider, DashboardMetadata metadata, DashboardState state) {
        app.get("/inspection/v1/overview", ctx -> ctx.json(DashboardSnapshotFactory.createOverview(inspectionProvider, metadata, state)));
        app.get("/inspection/v1/messages", ctx -> ctx.json(DashboardSnapshotFactory.createMessages(inspectionProvider, metadata)));
        app.get("/inspection/v1/consumers", ctx -> ctx.json(DashboardSnapshotFactory.createConsumers(inspectionProvider, metadata)));
        app.get("/inspection/v1/topology", ctx -> ctx.json(DashboardSnapshotFactory.createTopology(inspectionProvider, metadata)));
        app.get("/inspection/v1/queues", ctx -> ctx.json(DashboardSnapshotFactory.createQueues(inspectionProvider, metadata)));
        app.get("/inspection/v1/metrics", ctx -> ctx.json(DashboardSnapshotFactory.createMetrics(inspectionProvider, metadata, state)));
    }
}

File: src/TestApp/DashboardApi.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Routing;
using MyServiceBus.Inspection;

namespace TestApp;

public static class DashboardApi
{
    public static RouteGroupBuilder MapDashboardApi(
        this IEndpointRouteBuilder endpoints,
        DashboardMetadata metadata,
        string prefix = "/inspection/v1")
    {
        var group = endpoints.MapGroup(prefix)
            .WithTags("Dashboard");

        group.MapGet("/overview", (IBusInspectionProvider inspectionProvider, DashboardState state) =>
            TypedResults.Ok(DashboardSnapshotFactory.CreateOverview(inspectionProvider, metadata, state)));

        group.MapGet("/messages", (IBusInspectionProvider inspectionProvider) =>
            TypedResults.Ok(DashboardSnapshotFactory.CreateMessages(inspectionProvider, metadata)));

        group.MapGet("/consumers", (IBusInspectionProvider inspectionProvider) =>
            TypedResults.Ok(DashboardSnapshotFactory.CreateConsumers(inspectionProvider, metadata)));

        group.MapGet("/topology", (IBusInspectionProvider inspectionProvider) =>
            TypedResults.Ok(DashboardSnapshotFactory.CreateTopology(inspectionProvider, metadata)));

        group.MapGet("/queues", (IBusInspectionProvider inspectionProvider) =>
            TypedResults.Ok(DashboardSnapshotFactory.CreateQueues(inspectionProvider, metadata)));

        group.MapGet("/metrics", (IBusInspectionProvider inspectionProvider, DashboardState state) =>
            TypedResults.Ok(DashboardSnapshotFactory.CreateMetrics(inspectionProvider, metadata, state)));

        return group;
    }
}

public sealed record DashboardMetadata(string ServiceName, string TransportName);
